piwik-sdk/src/main/java/org/piwik/sdk/CustomVariables.java

```diff
@@ -46,12 +42,14 @@ public CustomVariables() {
      *              For example, if you choose to store the variable name = "Gender" in index = 1
      *              and you record another custom variable in index = 1, then the "Gender" variable
      *              will be deleted and replaced with the new custom variable stored in index 1.
+     *              You may configure Piwik to track more custom variables than 5.
+     *              Read more: http://piwik.org/faq/how-to/faq_17931/
      * @param name  of a specific Custom Variable such as "User type".
      * @param value of a specific Custom Variable such as "Customer".
      * @return super.put result if index in right range and name/value pair aren't null
      */
     public JSONArray put(int index, String name, String value) {
-        if (index > 0 && index <= MAX_VARIABLES && name != null & value != null) {
+        if (index > 0 && name != null & value != null) {
```

Comment:
Wrong indexes will be ignored on [the server side](https://github.com/piwik/piwik/blob/3d0684cda418d51596626034b8811942ed413f38/core/Tracker/Request.php#L574-L581)